File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.go
func totalNumbers(digits []int) int {
    // 3-digit even numbers
    seen := make([]bool, 1000)
    ans := 0
    for i := 0; i < len(digits); i++ {
        if digits[i] == 0 {
            continue
        }
        for j := 0; j < len(digits); j++ {
            if j == i {
                continue
            }
            for k := 0; k < len(digits); k++ {
                if k == i || k == j || digits[k]%2 != 0 {
                    continue
                }
                num := digits[i]*100 + digits[j]*10 + digits[k]
                if !seen[num] {
                    seen[num] = true
                    ans++
                }
            }
        }
    }

    return ans
}

// array, brute force
// time: O(n^3)
// space: O(1)

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        # 3-digit even numbers
        seen = [False] * 1000
        ans = 0
        for i in range(len(digits)):
            if digits[i] == 0:
                continue
            for j in range(len(digits)):
                if j == i:
                    continue
                for k in range(len(digits)):
                    if k == i or k == j or digits[k] % 2 != 0:
                        continue
                    num = digits[i] * 100 + digits[j] * 10 + digits[k]
                    if not seen[num]:
                        seen[num] = True
                        ans += 1
        return ans

# array, brute force
# time: O(n^3)
# space: O(1)